pkg/clusterbuilder/providers/capa/cluster.go

```diff
@@ -0,0 +1,85 @@
+package capa
+
+import (
+	_ "embed"
+	"fmt"
+	"math/rand"
+
+	"github.com/giantswarm/cluster-standup-teardown/pkg/values"
+
+	"github.com/giantswarm/clustertest/pkg/application"
+	"github.com/giantswarm/clustertest/pkg/organization"
+	"github.com/giantswarm/clustertest/pkg/utils"
+)
+
+var (
+	//go:embed values/cluster_values.yaml
+	baseClusterValues string
+	//go:embed values/default-apps_values.yaml
+	baseDefaultAppsValues string
+)
+
+// ClusterBuilder is the CAPA ClusterBuilder
+type ClusterBuilder struct{}
+
+// NewClusterApp builds a new CAPA cluster App
+func (c *ClusterBuilder) NewClusterApp(clusterName string, orgName string, clusterValuesFile string, defaultAppsValuesFile string) *application.Cluster {
+	if clusterName == "" {
+		clusterName = utils.GenerateRandomName("t")
+	}
+	if orgName == "" {
+		orgName = utils.GenerateRandomName("t")
+	}
+
+	return application.NewClusterApp(clusterName, application.ProviderAWS).
+		WithOrg(organization.New(orgName)).
+		WithAppValues(
+			values.MustOverlayValues(baseClusterValues, clusterValuesFile),
+			values.MustOverlayValues(baseDefaultAppsValues, defaultAppsValuesFile),
+			&application.TemplateValues{
+				ClusterName:  clusterName,
+				Organization: orgName,
+			},
+		)
+}
+
+var (
+	//go:embed values/private-cluster_values.yaml
+	basePrivateClusterValues string
+	//go:embed values/private-default-apps_values.yaml
+	basePrivateDefaultAppsValues string
+)
+
+// PrivateClusterBuilder is the private CAPA ClusterBuilder
+type PrivateClusterBuilder struct{}
+
+// NewClusterApp builds a new private CAPA cluster App
+func (c *PrivateClusterBuilder) NewClusterApp(clusterName string, orgName string, clusterValuesFile string, defaultAppsValuesFile string) *application.Cluster {
```

Comment:
just in case I'm missing something. is it actually possible to create a private cluster from the standup cli?

Last time I tried it needed some code changes. If it's a missing feature and you have it under the radar, please ignore :)

Reply:
It doesn't currently as it wasn't needed by any of the test pipelines.